pkg/grpc/server.go

```diff
@@ -149,6 +151,12 @@ func (s *server) getMiddlewareOptions() []grpc_go.ServerOption {
 		opts,
 		grpc_go.UnaryInterceptor(unaryServerInterceptor))
 
+	if s.authToken != "" {
+		opts = append(
+			opts,
+			grpc_go.UnaryInterceptor(setAPIAuthenticationMiddlewareUnary(s.authToken, auth.APITokenHeader)),
```

Comment:
this will panic if metric is enabled because grpc server allows to use only one `grpc_go.UnaryInterceptor` option. `setAPIAuthenticationMiddlewareUnary` needs to be in between L146 and L147. I would create array for all unary interceptors and create one unary interceptor chain.

Reply:
changed